### ISSUE TITLE
[Accessibility] CollectionView Selected on Android

### ISSUE THREAD

**Issue by tj-devel709** [dotnet/maui#29455]
### Description

This issue is related to: https://github.com/dotnet/maui/pull/29014.

When we select an item inside a CollectionView with Multiple as the SelectionMode, we want to set ItemView.Selected to true. However, we saw that when we unselect an item with the TalkBack accessibility feature on the android device, the TalkBack would immediately announce the top most selected item. This is not the wanted behavior. We would expect that the item would just be unselected and not read out a different item. 

[This commit here](https://github.com/dotnet/maui/pull/29014/commits/b74cb7a41dee12fb06304a8b592880910007c5e3) removes the changes from Android and the screenshots from the UITest that should be re-enabled after this is fixed.

### Steps to Reproduce

1. Create a CollectionView with SelectionMode set to Multiple.
2. On an android device, turn on TalkBack accessibility feature.
3. Select multiple items.
4. Unselect an item and hear the screen reader announce the top most selected item (incorrectly)

Alternatively, you can use the UITest Issue21375 in the HostApp to test on android and selecting the "Multiple" button and then selecting and unselecting items.

### Version with bug

9.0.60 SR6

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Relevant log output

```shell

```
